c++/src/kj/array.h

```diff
@@ -722,6 +853,24 @@ T* HeapArrayDisposer::allocateUninitialized(size_t count) {
 
 }
 
+template <typename T>
+T* SmallArrayDisposer::allocate(void* firstElement, size_t count) {
```

Comment:
These two functions are now largely just copies of their HeapArrayDisposer counterparts. In my earlier PR, I had them all share code -- but the code they shared was the template metaprogramming version of `if constexpr`, and it's much clearer to just duplicate that, now.